src/py/flwr/server/state/in_memory_state.py

```diff
@@ -183,6 +190,21 @@ def unregister_node(self, node_id: int) -> None:
             raise ValueError(f"Node {node_id} is not registered")
         self.node_ids.remove(node_id)
 
-    def get_nodes(self) -> Set[int]:
+    def get_nodes(self, workload_id: str) -> Set[int]:
         """Return all available client nodes."""
+        if workload_id not in self.workload_ids:
+            log(ERROR, "`workload_id` is invalid")
+            return set()
         return self.node_ids
+
+    def create_workload(self) -> str:
+        """Create one workload."""
+        # Create, store, and return workload ID
+        for _ in range(100):
```

Comment:
Let's remove this. We can just try it once, and raise an error if the workload_id already exists.